### PR TITLE
MASTERDIET-39

### PR DESCRIPTION
**Pull request by jfoder**

### CODE REVIEW

File: src/main/java/pl/agh/edu/master_diet/core/model/shared/Gender.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package pl.agh.edu.master_diet.core.model.shared;
+
+public enum Gender {
+    FEMALE(-161), MALE(5);
```
[the remark this answers is not in the document]
Comment (RafisSomeone) [reply src/main/java/pl/agh/edu/master_diet/core/model/shared/Gender.java:4]: it's needed to calculations

File: src/main/java/pl/agh/edu/master_diet/core/model/rest/userPlan/UserPlanResponse.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import lombok.Builder;
+import lombok.NonNull;
+
+@Builder
+public class UserPlanResponse {
+
+    @NonNull
+    private Long userId;
+    @NonNull
```
[the remark this answers is not in the document]
Comment (jfoder) [reply src/main/java/pl/agh/edu/master_diet/core/model/rest/userPlan/UserPlanResponse.java:20]: Done

File: src/main/java/pl/agh/edu/master_diet/core/model/rest/userPlan/UserPlanResponse.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package pl.agh.edu.master_diet.core.model.rest.userPlan;
+
+import lombok.Builder;
+import lombok.NonNull;
+
+@Builder
+public class UserPlanResponse {
```
[the remark this answers is not in the document]
Comment (jfoder) [reply src/main/java/pl/agh/edu/master_diet/core/model/rest/userPlan/UserPlanResponse.java:15]: Done

File: src/main/java/pl/agh/edu/master_diet/core/model/shared/Gender.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+    Gender(int value) {
```
[the remark this answers is not in the document]
Comment (jfoder) [reply src/main/java/pl/agh/edu/master_diet/core/model/shared/Gender.java]: Done

File: src/main/java/pl/agh/edu/master_diet/core/model/shared/Gender.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfoder) [reply src/main/java/pl/agh/edu/master_diet/core/model/shared/Gender.java:9]: I leave it as it is, maybe @RafisSomeone will change this enum on his branch

File: src/main/java/pl/agh/edu/master_diet/core/model/database/Achievement.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "achievement")
+@Getter
+@Setter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class Achievement {
```
[the remark this answers is not in the document]
Comment (jfoder) [reply src/main/java/pl/agh/edu/master_diet/core/model/database/Achievement.java:24]: We should avoid bidirectional relations and add them only if it's necessary

File: src/main/java/pl/agh/edu/master_diet/core/model/database/UserPlan.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    private Goal goal;
+
+    @Column(nullable = false)
+    private Float currentWeight;
+
+    @OneToOne
+    @MapsId
+    private User user;
+
+    public UserPlanResponse map2UserPlanResponse() {
```
[the remark this answers is not in the document]
Comment (jfoder) [reply src/main/java/pl/agh/edu/master_diet/core/model/database/UserPlan.java]: Moved to `UserPlanConverter`

**Pull request merged**